[ADD] Sales: Members Backport (#10158)
odoo/documentation: content/applications/sales/members.rst

=======
Members
=======

The **Members** application is where all operations related to memberships can be configured and
managed. The **Members** app integrates with the **Sales** and **Accounting** applications to sell
and invoice memberships directly to customers.

Membership products
===================

To create a new membership product, navigate to :menuselection:`Members app --> Configuration -->

> Membership Products`, and click :guilabel:`New` to open a blank product record.

Reviewer: ```suggestion
Membership Products`, and click :guilabel:`New` to open a blank product form.
```

On the new product form, enter a name for the product in the field under :guilabel:`Product Name`.
Use the two blank fields to the right of :guilabel:`Membership Duration` to set a date from which

> membership becomes active. Finally, use the :guilabel:`Membership Fee` field to set the price at

Reviewer: ```suggestion
the membership becomes active. Finally, use the :guilabel:`Membership Fee` field to set the price at
```

which the product is sold to customers.

The following optional fields can also be configured for each product:

An internal reference number can be set in the field to the right of :guilabel:`Internal Reference`,

> and a category set using the dropdown to the right of :guilabel:`Product Category`. Set a company

Reviewer: ```suggestion
and a category set using the drop-down menu to the right of :guilabel:`Product Category`. Set a company
```
Drop-down is always hyphenated and followed by either "menu" or "window" -- depending on the circumstance

using the field to the right of :guilabel:`Company` for the product to appear in that company's
inventory. Use the slider to the right of :guilabel:`Active` to set hide a product without removing

> it.

Reviewer: please rework/reword this portion about 'Active'

Tick the checkbox next to :guilabel:`Visible on current website` if desired, and use the field
under :guilabel:`Description` for an internal product description, and :guilabel:`Sales Description`
for a product description that will be visible on quotations. Use the field to the right of
:guilabel:`Income Account` to use a different value than the default from the product category.
Finally, set the default tax used when selling this product using the field next to

> :guilabel:`Taxes`.

Reviewer: While this info is correct, I feel like most docs with a section like this tend use a list format.

[name of field] : Explanation of the field.

That tends to be well received... so, consider that during your revisions.

The way it is currently written doesn't rely the information as cleanly (or as efficiently) as possible

Reviewer: Upon further thought, I think you should *definitely* rework/reformat/reword this section so it is a cleaner, more digestible list. Doing so will add to the overall value of the doc. Thanks!

.. note::
   Membership products require a start and end date, as they are used to determine :ref:`membership
   status <sales/membership-status>`. Membership products can be sold *before* their active start
   date.

.. image:: members/membership-product.png
   :align: center
   :alt: A new membership product in the members app.

Membership products can be added to a sales order, and invoiced as regular products or
subscriptions.

Activate a membership
=====================

To activate a membership from the **Contacts** application, navigate to the :menuselection:`Contacts
app`, and click on a contact to open that specific contact form.

From the contact form, open the :guilabel:`Membership` tab, and click :guilabel:`Buy
Membership`.

.. tip::
   To successfully buy a membership, both the :guilabel:`Account Receivable` and :guilabel:`Account

>    Payable` lines must first be configured for that contact. These can be found in the contact

Reviewer: ```suggestion
   Payable` lines **must** be configured for that contact. These can be found in the contact
```

=======
Members
=======

> The *Members* application is where all operations related to memberships can be configured and

Reviewer: ```suggestion
The **Members** application is where all operations related to memberships can be configured and
```
using **BOLD** for app names now, based on most recent RST guideline directives

Reviewer: Please implement this suggestion, as it is now a requirement when it comes to formatting application names. Thanks

> managed. The *Members* app integrates with the *Sales* and *Accounting* applications to sell and

Reviewer: ```suggestion
managed. The **Members** app integrates with the **Sales** and **Accounting** applications to sell and
```

invoice memberships directly to customers.

Membership products
===================

To create a new membership product, navigate to :menuselection:`Members app --> Configuration -->
Membership Products`, and click :guilabel:`New` to open a blank product record.

Complete the blank form with the necessary information, including the :guilabel:`Membership

> Duration`.

Reviewer: I think this needs to be expanded upon, and provide the reader with information about all the fields present on the form. Doing so would provide some much-needed context, and add some value to the overall doc.

.. note::
   Membership products require a start and end date, as they are used to determine :ref:`membership
   status <sales/membership-status>`. Membership products can be sold *before* their active start
   date.

.. image:: members/membership-product.png
   :align: center
   :alt: A new membership product in the members app.

Membership products can be added to a sales order, and invoiced as regular products or
subscriptions.

Activate a membership
=====================

> To activate a membership from the *Contacts* application, navigate to :menuselection:`Contacts app`,

Reviewer: ```suggestion
To activate a membership from the **Contacts** application, navigate to the :menuselection:`Contacts app`,
```

> and click on a contact to open that specific contact's detail form.

Reviewer: ```suggestion
and click on a contact to open that specific contact form.
```

> From the resulting contact form, open the :guilabel:`Membership` tab, and click :guilabel:`Buy

Reviewer: ```suggestion
From the contact form, open the :guilabel:`Membership` tab, and click :guilabel:`Buy
```

Membership`.

.. tip::
   To successfully buy a membership, both the :guilabel:`Account Receivable` and :guilabel:`Account
   Payable` lines must first be configured for that contact. These can be found in the contact

>    detail form in the :guilabel:`Accounting` tab under the :guilabel:`Accounting entries` heading.

Reviewer 2: QQ: this section is not present in the `17.0` version of this doc, was that on purpose?

Author: Yes, when following the flow for 16, I found it was necessary to have these 2 fields configured before buying a membership to prevent an error.

>    detail form in the :guilabel:`Accounting` tab under the :guilabel:`Accounting entries` heading.

Reviewer: ```suggestion
   form's :guilabel:`Accounting` tab, under the :guilabel:`Accounting entries` section.
```

the drop-down menu. Then, configure a :guilabel:`Member Price`.

Click :guilabel:`Invoice Membership` when both fields are filled in. Doing so reveals a
:guilabel:`Membership Invoices` page, wherein invoices can be confirmed and completed.

Alternatively, to offer a free membership, tick the :guilabel:`Free Member` checkbox, in the
:guilabel:`Membership` tab of a contact form.

.. _sales/membership-status:

Membership status
=================

The :guilabel:`Current Membership Status` is listed on the :guilabel:`Membership` tab of each
contact record:

- :guilabel:`Non Member`: a partner who has **not** applied for membership.
- :guilabel:`Cancelled Member`: a member who has cancelled their membership.
- :guilabel:`Old Member`: a member whose membership end date has passed.
- :guilabel:`Waiting Member`: a member who has applied for membership, but whose invoice has not
  yet been created.
- :guilabel:`Invoiced Member`: a member whose invoice has been created, but has not been paid.
- :guilabel:`Paid Member`: a member who has paid the membership fee.

Publish members directory
=========================

> To publish a list of active members on the website, the *Online Members Directory* application must

Reviewer: ```suggestion
To publish a list of active members on the website, the **Online Members Directory** module must
```

Reviewer: please implement

first be :ref:`installed <general/install>`. After installing the module, add the `/members` page to
the website's menu by :doc:`editing the website menu <../websites/website/pages/menus>`.

.. image:: members/membership-directory-app.png
   :align: center
   :alt: The Online Members directory module in Odoo.

Publish individual members
--------------------------

> Return to :menuselection:`CRM app --> Sales --> Customers`, and click the Kanban card for a member.

Reviewer: ```suggestion
Navigate to :menuselection:`CRM app --> Sales --> Customers`, and click the Kanban card for a member.
```
